frontend/trpc/routes/paymentManagement.ts

```
@@ -0,0 +1,280 @@
+
+export const paymentManagementRouter = createRouter({
+  // Get payment status for a dividend round
+  getDividendPaymentStatus: protectedProcedure
+    .input(z.object({
+      companyId: z.string(),
+      dividendRoundId: z.number(),
+    }))
+    .query(async ({ input, ctx }) => {
+      try {
+        const response = await fetch(
+          `${process.env.BACKEND_URL}/internal/companies/${ctx.company!.externalId}/dividend_rounds/${input.dividendRoundId}/payment_status`,
```

ESLint

[error] Forbidden non-null assertion.

(@typescript-eslint/no-non-null-assertion)